### ISSUE TITLE
[client/catapult] Missing resolved address information when sending > 1 transaction to same address alias in one block

### ISSUE THREAD

**Issue by cryptoBeliever** [symbol/symbol#515]
### System Information

- **Client Version:** 1.0.3.5
- **REST Version:** 2.4.2
- **Operating System:** Linux
- **Installed With:** Bootstrap
- **VPS Provider:** Other
- **Are you fully synchronized?** Yes
- **Did you try to restart your node?** Yes

#### Description

Actual behavior:
In case when we're sending > 1 transaction to the same namespace address alias and transactions are included in one block then only one of them has resolved namespace address in `meta.addresses`.

Expected behavior:
All transactions sent to the namespace address alias should have resolved the namespace address in `meta.addresses`

#### Steps to Reproduce

[1.] Create account A and refill it with tokens
[2.] Create namespace N.
[3.] Set namespace N as an address alias for A
[4.] Send > 1 transactions to alias A so that all transactions are included in one block.

Result: Only one transaction has resolved the namespace alias address in `meta.addresses` field in the mongo database.

#### Example reproduced cases

The problem was reproduced on test net in blocks 135795, 136750, and 136753 (https://testnet.symbol.fyi/blocks/136753).

**Comment by Jaguar0625** [comment]
Can you add the contents of `meta.addresses` for each transaction?

**Comment by cryptoBeliever** [comment]
Example from block 135795 pasted by Wayon on Discord.:

```
> db.transactions.find({"meta.height": 135795}).forEach(printjson)
{
    "_id" : ObjectId("63b90943fe30f382260d09b6"),
    "meta" : {
        "height" : NumberLong(135795),
        "hash" : BinData(0,"O8wCErfRpojr2oJrh6nGuHK1OgTCzBy21g9Dohus7a8="),
        "merkleComponentHash" : BinData(0,"O8wCErfRpojr2oJrh6nGuHK1OgTCzBy21g9Dohus7a8="),
        "index" : 0,
        "addresses" : [
            BinData(0,"mKSjI/JEK6Aqh0vA/uBzksZMUCqU0Xjn"),
            BinData(0,"mX952aAJlJCSAAAAAAAAAAAAAAAAAAAA"),
            BinData(0,"mAjcD/+6PZ2Zl6GvdA41LZfbzNA7qGkv")
        ]
    },
    "transaction" : {
        "size" : 335,
        "signature" : BinData(0,"4bSvL5y11F8vWCeJB4SSwPjg3W5wmpPE+ywDs0HFon3j0f5iJQJL5Z4I6v9CCRPo9AvfqL5I4rnD7z8+mGStBw=="),
        "signerPublicKey" : BinData(0,"4d0Hlk89l9HqT03WVVtI9NkNipCI19WXB88Tiu0UpRg="),
        "version" : 1,
        "network" : 152,
        "type" : 16724,
        "maxFee" : NumberLong(33500),
        "deadline" : NumberLong("5827625496"),
        "recipientAddress" : BinData(0,"mX952aAJlJCSAAAAAAAAAAAAAAAAAAAA"),
        "message" : BinData(0,"AAogICAg4pGgM+WbnuebrgogICAg4pGh77yX44GP44KJ44GE44Gn44GX44Gf44O744O7CiAgICDikaLjga/jgYQKICAgIOKRo+OBr+OBhAogICAg4pGk5LuK5Zue44Gh44KH44Gj44Go5pep44GL44Gj44Gf5rCX44GM44GX44G+44GZ44CC5YWo5Yqb44Gn44GM44KT44Gw44KK44G+44GX44Gf56yRICAKICAgIA=="),
        "mosaics" : [ ]
    }
}
{
    "_id" : ObjectId("63b90943fe30f382260d09b7"),
    "meta" : {
        "height" : NumberLong(135795),
        "hash" : BinData(0,"nc23Mcao9B9l/67GaHqLnRc5C74FlpZ1ZwKbGeOvWYQ="),
        "merkleComponentHash" : BinData(0,"nc23Mcao9B9l/67GaHqLnRc5C74FlpZ1ZwKbGeOvWYQ="),
        "index" : 1,
        "addresses" : [
            BinData(0,"mX952aAJlJCSAAAAAAAAAAAAAAAAAAAA"),
            BinData(0,"mN04Jo9BP8xxgyzjdy/61j1reoKpW0zI")
        ]
    },
    "transaction" : {
        "size" : 352,
        "signature" : BinData(0,"CLfZ4nvm+AKRXsENc9GMP37JXiy/aBW0MxaaTPOrSpxYV7Y+kHjNJI/Cu56jKoZ0/yKJMNFM5ILzjx00lVemBA=="),
        "signerPublicKey" : BinData(0,"8cwlxd4qfWKI+DaDSkQvp+DuSiwZuXeuXPqsfvK7LLU="),
        "version" : 1,
        "network" : 152,
        "type" : 16724,
        "maxFee" : NumberLong(35200),
        "deadline" : NumberLong("5827643427"),
        "recipientAddress" : BinData(0,"mX952aAJlJCSAAAAAAAAAAAAAAAAAAAA"),
        "message" : BinData(0,"AAogICAg4pGg77yS5Zue55uuCiAgICDikaHvvJHvvJAKICAgIOKRouOBr+OBhAogICAg4pGj44Gv44GECiAgICDikaTjgYLjgorjgYzjgajjgYbjgZTjgZbjgYTjgb7jgZnjgILku4rlm57jgoLmpb3jgZfjgY/lrabjgbnjgb7jgZfjgZ/jgILjgb7jgZ/mrKHlm57jgoLlj4LliqDjgZfjgZ/jgYTjgajmgJ3jgYTjgb7jgZnjgIIgIAogICAg"),
        "mosaics" : [ ]
    }
}
```

**Comment by Jaguar0625** [comment]
is this only an issue when 0 mosaics are transferred?

**Comment by cryptoBeliever** [comment]
> is this only an issue when 0 mosaics are transferred?

It's also when mosaics are not empty. Example in block 141182

```
> db.transactions.find({"meta.height": 141182}).forEach(printjson)

{
	"_id" : ObjectId("63bc4202de9cc4e723066afb"),
	"meta" : {
		"height" : NumberLong(141182),
		"hash" : BinData(0,"YFyp7wEftiy1Nk6APCY18RNrzLKgELxqYqDDfYaTMyI="),
		"merkleComponentHash" : BinData(0,"YFyp7wEftiy1Nk6APCY18RNrzLKgELxqYqDDfYaTMyI="),
		"index" : 0,
		"addresses" : [
			BinData(0,"mKRR/nhxQWunSA/jOPCLzW19ax2D4Lao"),
			BinData(0,"mUMb5H3q0q+VAAAAAAAAAAAAAAAAAAAA"),
			BinData(0,"mI6NdawGf7OmYYkjCLp+wowJ43KpBRff")
		]
	},
	"transaction" : {
		"size" : 192,
		"signature" : BinData(0,"lwhg77LRJ6iUEqiJJUxGXKogRgac7xX1c2bGJKixTIEUKozOt0q1fm+RPOFcutdxsahd9tGXmP4Y0vSy9VNCAg=="),
		"signerPublicKey" : BinData(0,"CqJ8YFsMNXyUwWeNgw8lQk7DkiFV5VPfOyBmRC72wMM="),
		"version" : 1,
		"network" : 152,
		"type" : 16724,
		"maxFee" : NumberLong(19200),
		"deadline" : NumberLong("6038747869"),
		"recipientAddress" : BinData(0,"mUMb5H3q0q+VAAAAAAAAAAAAAAAAAAAA"),
		"mosaics" : [
			{
				"id" : NumberLong("7428778700120476043"),
				"amount" : NumberLong(2)
			},
			{
				"id" : NumberLong("8268645399043017678"),
				"amount" : NumberLong(1000000)
			}
		]
	}
}
{
	"_id" : ObjectId("63bc4202de9cc4e723066afa"),
	"meta" : {
		"height" : NumberLong(141182),
		"hash" : BinData(0,"MbK8hTtrs27OzWsPjERH0KDNIEgYUiL0GF8bPcqutdw="),
		"merkleComponentHash" : BinData(0,"MbK8hTtrs27OzWsPjERH0KDNIEgYUiL0GF8bPcqutdw="),
		"index" : 1,
		"addresses" : [
			BinData(0,"mUMb5H3q0q+VAAAAAAAAAAAAAAAAAAAA"),
			BinData(0,"mI6NdawGf7OmYYkjCLp+wowJ43KpBRff")
		]
	},
	"transaction" : {
		"size" : 192,
		"signature" : BinData(0,"E5N8mrcpc4qge2mg3iiQpVzIJjsMIng4QvngB2oH89ByRbUoPYIrauEWuTdlHx2kiGd7BqobgUc7F0MJygh9Bw=="),
		"signerPublicKey" : BinData(0,"CqJ8YFsMNXyUwWeNgw8lQk7DkiFV5VPfOyBmRC72wMM="),
		"version" : 1,
		"network" : 152,
		"type" : 16724,
		"maxFee" : NumberLong(19200),
		"deadline" : NumberLong("6038754663"),
		"recipientAddress" : BinData(0,"mUMb5H3q0q+VAAAAAAAAAAAAAAAAAAAA"),
		"mosaics" : [
			{
				"id" : NumberLong("7428778700120476043"),
				"amount" : NumberLong(3)
			},
			{
				"id" : NumberLong("8268645399043017678"),
				"amount" : NumberLong(2000000)
			}
		]
	}
}
```

**Comment by Jaguar0625** [comment]
Fixed in 494570ac0ef7447616993023cd3c761b92ab0b17.

**Comment by cryptoBeliever** [comment]
After the fix alias is correctly resolved:
https://401-sai-dual.symboltest.net:3001/transactions/confirmed?height=368466
https://401-sai-dual.symboltest.net:3001/transactions/confirmed?address=TCJVHYTL32T4KOHSU55I6EPLWMAJCFXRNRCVJXA&height=368466

In case when we have 2 transactions sent to alias in one block both are returned to resolved (by namespace) alias :heavy_check_mark:
